### PR TITLE
chore: extract common sections across app types into separate dir

### PR DESCRIPTION
**Pull request by efekarakus**
Part 1 of #833

The files under `templates/applications/common/cf/` are sections that should be shared across all app types. This will allow us to DRY our templates. We should be able to write new app type templates faster, remove the chance of errors while copy pasting, and keep things consistent across applications.

Please let me know if you have any disagreements, also if you think there are other sections that can be extracted to the `common/cf` dir.

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by efekarakus** [comment]
- The diffed files are the same (minus some unnecessary new lines).
- ✅ Ran `app deploy` with secrets, env vars, and addons and it worked fine.

File: internal/pkg/deploy/cloudformation/stack/lb_fargate_app.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func (c *LBFargateStackConfig) Template() (string, error) {
 	if err != nil {
 		return "", err
 	}
-	content, err := c.parser.Parse(lbFargateAppTemplatePath, struct {
+	content, err := c.parser.ParseAppTemplate("lb-web-app", struct {
```
Comment (iamhopaul123) [inline internal/pkg/deploy/cloudformation/stack/lb_fargate_app.go:100]: Why not use `lbFargateAppTemplateName`?

Comment (efekarakus) [reply internal/pkg/deploy/cloudformation/stack/lb_fargate_app.go]: I forgot lol, fixed thanks!

File: internal/pkg/template/template.go [hunk 2]
```diff
@@ -92,10 +130,48 @@ func (c *Content) MarshalBinary() ([]byte, error) {
...
+
 func (t *Template) read(path string) (string, error) {
 	s, err := t.box.FindString(path)
 	if err != nil {
 		return "", fmt.Errorf("read template %s: %w", path, err)
 	}
 	return s, nil
 }
+
+func (t *Template) parseTemplate(name, path string, options ...ParseOption) (*template.Template, error) {
```
Comment (bvtujo) [inline internal/pkg/template/template.go:162]: Does this method need to be attached to the Template struct?  I see that it needs the reference to `t` in the call to `read()` but is there any drawback to generating a new Template within a method attached to a Template? 

Or is this just semantic confusion on my part between the "template parser" and the "parsed template?"

File: internal/pkg/template/template.go [hunk 3]
```diff
@@ -59,23 +84,36 @@ func (t *Template) Read(path string) (*Content, error) {
 
 // Parse parses the template under "/templates/{path}" with the specified data object and returns its content.
 func (t *Template) Parse(path string, data interface{}, options ...ParseOption) (*Content, error) {
-	content, err := t.read(path)
+	tpl, err := t.parseTemplate("template", path, options...)
```
Comment (bvtujo) [inline internal/pkg/template/template.go:87]: Why is the `param` hardcoded to "template" here? I don't see it having much effect through the chain, but is this just necessary for the call to template.New()?

Comment (efekarakus) [reply internal/pkg/template/template.go:87]: > but is this just necessary for the call to template.New()?

Correct! The reason why they require a name I believe is for the scenario below with `ParseAppTemplate`.  
To illustrate, let's say I created a new Template object named "base" with:
```go
tpl := template.New("base").Parse(content)
```
where the content contains two nested templates "hello" and "world".
```go
{{define "hello"}}Hello, World!{{end}}
{{define "bye"}}Bye, World!{{end}}
```
Then I can decide whether to execute the entire template with `tpl.Execute(buf, data)` or I can specify a particular nested template with `tpl.ExecuteTemplate(buf, "hello", data)` that will render only "Hello, World!".

So yeah, if your named template won't be used anywhere else it's useless. But if you need to include it into another template then the name becomes important.

File: internal/pkg/template/template.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/template/template.go:162]: ah yeah this is definitely confusing lol. So `t` refers to our own `Template` object but `newTemplate` returns a `text/template` object 😛 so there are two different `Template` types in this pkg.

I'll rename the methods this way, do you think it makes it clearer?  

`parseTemplate -> parse` and add the comment `// parse reads the file at path and return a text/template object with the given name`

 `newTemplate -> newTextTemplate`

Comment (bvtujo) [reply internal/pkg/template/template.go:162]: Yes, that clarifies things nicely. I think in this ambiguous case it's good to disambiguate by using the short name for the object defined in the current file or package and the longer name for external objects with the same name.

File: internal/pkg/template/template.go [hunk 4]
```diff
@@ -59,23 +84,36 @@ func (t *Template) Read(path string) (*Content, error) {
...
-	content, err := t.read(path)
+	tpl, err := t.parse("template", path, options...)
 	if err != nil {
 		return nil, err
 	}
-
-	tpl := template.New("template")
-	for _, opt := range options {
-		tpl = opt(tpl)
+	buf := &bytes.Buffer{}
```
Comment (iamhopaul123) [inline internal/pkg/template/template.go:91]: maybe `bytes.NewBuffer(nil)` is a better way to initialize a buffer? Just a nit though.

Comment (efekarakus) [reply internal/pkg/template/template.go:91]: Done!

**Pull request merged**